fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb

```diff
@@ -0,0 +1,116 @@
+require 'fluent/plugin/input'
+require 'fluent/plugin/in_monitor_agent'
+require 'fluent/plugin/prometheus'
+
+module Fluent::Plugin
+  # Collect log_collected_bytes_total metric from in_tail plugin.
+  class CollectedTailMonitorInput < Fluent::Plugin::Input
+    Fluent::Plugin.register_input('collected_tail_monitor', self)
+    include Fluent::Plugin::PrometheusLabelParser
+
+    helpers :timer
+
+    #ideally interval must be rotatewait > 0 ? rotatewait/2 : 1.0 and if interval >= rotate_wait we should log a warning. For rotatewait = 5 sec, interval can be set to 2 sec
+    config_param :interval, :time, default: 2
+    attr_reader :registry
+
+    MONITOR_IVARS = [
+      :tails,
+    ]
+
+    def initialize
+      super
+      @registry = ::Prometheus::Client.registry
+    end
+
+    def multi_workers_ready?
+      true
+    end
+
+    def configure(conf)
+      super
+      hostname = Socket.gethostname
+      expander_builder = Fluent::Plugin::Prometheus.placeholder_expander(log)
+      expander = expander_builder.build({ 'hostname' => hostname, 'worker_id' => fluentd_worker_id })
+      @base_labels = parse_labels_elements(conf)
+      @base_labels.each do |key, value|
+        unless value.is_a?(String)
+          raise Fluent::ConfigError, "record accessor syntax is not available in collected_tail_monitor"
+        end
+        @base_labels[key] = expander.expand(value)
+      end
+
+      if defined?(Fluent::Plugin) && defined?(Fluent::Plugin::MonitorAgentInput)
+        # from v0.14.6
+        @monitor_agent = Fluent::Plugin::MonitorAgentInput.new
+      else
+        @monitor_agent = Fluent::MonitorAgentInput.new
+      end
+    end
+
+    def start
+      super
+
+      @metrics = {
+        total_bytes_collected: get_counter(
+          :log_collected_bytes_total,
+          'Total bytes collected from file.'),
+      }
+      timer_execute(:in_collected_tail_monitor, @interval, &method(:update_monitor_info))
+    end
+
+
+    def update_monitor_info
+      opts = {
```

Comment:
This does not change which means we should declare this as an instance var "@opts" that we can use repeatedly

Reply:
This plugin implementation is derived from fluent-prometheus-plugin. There in_prometheus_tail_monitor doesn't use opt as instance var but local variable. :tail will be keeping track of all log files at any instant of time that this variable get assigned to.